### PR TITLE
HPCC-17671 Alphabetize users in add users to group

### PR DESCRIPTION
**Pull request by kunalaswani**
Alphabetizing users in groups when adding to a Group.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by kunalaswani** [comment]
@jeclrsg couple issues i noticed, it was refering to (Property 'username' does not exist on type 'unknown'.ts) and also the jira requested a change to the dropdown for multi selection of users... can you provide some input, i was unable to get mutliselect to load

File: esp/src/src-react/components/forms/Fields.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ import { joinPath } from "src/Utility";
 import nlsHPCC from "src/nlsHPCC";
 import { useBuildInfo, useLogicalClusters } from "../../hooks/platform";
 import { useContainerNames } from "../../hooks/cloud";
+import AsyncSelect from 'react-select/async';
+import { ColourOption, colourOptions } from '../data';
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/Fields.tsx:15]: These seem unrelated, so should both be removed?

File: esp/src/src-react/components/forms/Fields.tsx [hunk 2]
```diff
@@ -677,18 +679,17 @@ export interface GroupMembersProps extends Omit<AsyncDropdownProps, "options"> {
...
-                        key: user.username,
-                        text: user.username
-                    };
-                }) || [];
+            const usersArray = GroupMemberEditInputResponse?.Users?.User || [];
+            const _users = Array.from(usersArray, user => ({
+                key: user.username,
+                text: user.username
+            }));
+            _users.sort((a, b) => a.text.localeCompare(b.text));
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/Fields.tsx:692]: The sort() can be chained from the map()

<pre>
.map(user => {
  ...
}).sort(...)
|| [];
</pre>

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@kunalaswani There is a DropdownMulti component, but we don't have an async multiselect dropdown yet. That could be a bit complicated to implement.

File: esp/src/src-react/components/forms/Fields.tsx (hunk 2 above)
Comment (kunalaswani) [reply esp/src/src-react/components/forms/Fields.tsx]: Updated, and tested.

**Review by jeclrsg — APPROVED** [review]
